9.js

```diff
@@ -0,0 +1,70 @@
+const findMatchingParenIndex = (string, inputIndex) => {
+  let level = 1
+  let index = inputIndex
+
+  if (string.charAt(index) === '(') {
```

Comment:
Seeing all the `'('` and `')'`, I see an opportunity to refactor to constants to get rid of the magic strings.  This would allow us to avoid possible typo bugs, and make the characters stand out a little more in the code:
```js
const LEFT_PAREN = '('
const RIGHT_PAREN = ')'
// etc.
```